### PR TITLE
Small bugfixes and improvements

### PR DESCRIPTION
**Pull request by Pankraty**
After I ran a code analysis tool I found a few issues - some are more important, others are less.

The obvious bug here is fixed with commit e9c22cec85cbc17ddbde31e526a0ffb35cb29676. Aside from this,  some checks were reorganized (removed, added or added), ```IDisposable``` implemented for classes containing disposable members, other small enhancements made.

### CODE REVIEW

File: ClosedXML/Excel/XLWorkbook_Load.cs [hunk 1]
```diff
@@ -477,7 +477,7 @@ private void LoadSpreadsheetDocument(SpreadsheetDocument dSpreadsheet)
                                 continue;
                         }
 
-                        if (target != null && source != null)
+                        if (source != null)
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Load.cs]: I assume the static analysis tool said that `target` cannot be `null` here. So, for checks like these, I prefer to keep the check, to future-proof it. It's possible that something happens in the future that changes code higher up and then we would need the check. Rather than introducing a possible regression in the future, I'd prefer to keep this check her. It doesn't hurt to keep it.

File: ClosedXML/Excel/XLWorkbook_Load.cs [hunk 2]
```diff
@@ -1361,12 +1361,9 @@ private static void ParseReference(string item, out string sheetName, out string
                 else
                     formula = cell.CellFormula.Text;
 
-                if (cell.CellFormula.Reference != null)
-                {
-                    // Parent cell of shared formulas
-                    // Child cells will use this shared index to set its R1C1 style formula
-                    xlCell.FormulaReference = ws.Range(cell.CellFormula.Reference.Value).RangeAddress;
-                }
+                // Parent cell of shared formulas
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Load.cs]: Why did you remove this `if`?

File: ClosedXML/Excel/XLWorkbook_Load.cs [hunk 3]
```diff
@@ -1956,47 +1953,47 @@ private static void LoadAutoFilterColumns(AutoFilter af, XLAutoFilter autoFilter
 
                         if (xlDateGroupFilter.DateTimeGrouping >= XLDateTimeGrouping.Year)
                         {
-                            if (dateGroupItem?.Year?.HasValue ?? false)
+                            if (dateGroupItem.Year?.HasValue ?? false)
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Load.cs]: You don't have to change this one, but again, including the `?` is my personal preference to future-proof the code.

File: ClosedXML/Excel/XLWorkbook_Save.cs [hunk 4]
```diff
@@ -3476,7 +3477,7 @@ private void AddDifferentialFormats(WorkbookStylesPart workbookStylesPart, SaveC
             var differentialFormat = new DifferentialFormat();
 
             var diffFont = GetNewFont(new FontInfo { Font = cf.Style.Font as XLFont }, false);
-            if (diffFont?.HasChildren ?? false)
+            if (diffFont.HasChildren)
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Save.cs]: Definitely keep the `?` and `??`, please. I have been working on branches that might return `null`s. Not sure if I'm going to merge them, but again, I'd rather keep the code future-proof. Please revert these changes for all the "style" variables in these methods.

File: ClosedXML/XLHelper.cs [hunk 5]
```diff
@@ -116,8 +116,10 @@ internal static int TrimRowNumber(int rowNumber)
 
         public static bool IsValidColumn(string column)
         {
+            if (String.IsNullOrWhiteSpace(column))
+                return false;
             var length = column.Length;
-            if (String.IsNullOrWhiteSpace(column) || length > 3)
+            if (length > 3)
```
Comment (igitur) [inline ClosedXML/XLHelper.cs]: Probably not even necessary for a `length` variable?

File: ClosedXML/Excel/Rows/XLRow.cs [hunk 6]
```diff
@@ -336,7 +336,7 @@ public IXLRow AdjustToContents(Int32 startColumn, Int32 endColumn, Double minHei
                         else
                         {
                             Double rotation;
-                            if (textRotation == 90 || textRotation == 180 || textRotation == 255)
+                            if (textRotation == 90 || textRotation == 180)
```
Comment (igitur) [inline ClosedXML/Excel/Rows/XLRow.cs]: Actually, I have no idea why there is a special case for `textRotation == 255` a few lines back.

File: ClosedXML/Excel/AutoFilters/XLAutoFilter.cs [hunk 7]
```diff
@@ -116,62 +116,59 @@ public XLAutoFilter Sort(Int32 columnToSortBy, XLSortOrder sortOrder, Boolean ma
...
-            if (Enabled)
-            {
                 using (var rows = Range.Rows(2, Range.RowCount()))
                 {
                     foreach (IXLRangeRow row in rows)
                         row.WorksheetRow().Unhide();
                 }
 
-                foreach (var kp in Filters)
+            foreach (var kp in Filters)
```
Comment (igitur) [inline ClosedXML/Excel/AutoFilters/XLAutoFilter.cs]: Fix indentation of the preceding block.

File: ClosedXML/Excel/AutoFilters/XLAutoFilter.cs [hunk 8]
```diff
@@ -116,62 +116,59 @@ public XLAutoFilter Sort(Int32 columnToSortBy, XLSortOrder sortOrder, Boolean ma
             SortColumn = columnToSortBy;
 
             // Recalculate shown / hidden rows
-            if (Enabled)
```
Comment (igitur) [inline ClosedXML/Excel/AutoFilters/XLAutoFilter.cs]: Why did you remove this? It seems like a good check to have?

Comment (Pankraty) [reply ClosedXML/Excel/AutoFilters/XLAutoFilter.cs]: At the top of the method, there is already 
`if (!Enabled) throw new InvalidOperationException("Filter has not been enabled.");`

Comment (igitur) [reply ClosedXML/Excel/AutoFilters/XLAutoFilter.cs]: ah ok

File: ClosedXML/Excel/XLWorkbook_Load.cs (hunk 1 above)
Comment (Pankraty) [reply ClosedXML/Excel/XLWorkbook_Load.cs]: OK, got it. Will restore it.

File: ClosedXML/Excel/XLWorkbook_Load.cs (hunk 2 above)
Comment (Pankraty) [reply ClosedXML/Excel/XLWorkbook_Load.cs]: Here we are already inside the condition 
`if (cell.CellFormula != null && cell.CellFormula.SharedIndex != null && cell.CellFormula.Reference != null)`

Comment (igitur) [reply ClosedXML/Excel/XLWorkbook_Load.cs]: It would benefit me to expand the preceding code blocks! :-)

File: ClosedXML/XLHelper.cs (hunk 5 above)
Comment (Pankraty) [reply ClosedXML/XLHelper.cs]: It is used in the code below so I think it's OK to leave it.

File: ClosedXML/Excel/Rows/XLRow.cs (hunk 6 above)
Comment (Pankraty) [reply ClosedXML/Excel/Rows/XLRow.cs]: `textRotation == 255` stands for a vertical orientation, so it is natural to determine a height for this case independently.
image
I wonder if this special case was ever tested?..

**Comment by igitur** [comment]
Thanks!
